doc/sphinx/src/outputs.rst

```diff
@@ -158,6 +161,134 @@ This will produce a text file (``.hst``) output file every 1 units of
 simulation time. The content of the file is determined by the functions
 enrolled by a specific package, see :ref:`state history output`.
 
+Histograms
+----------
+
+Parthenon supports calculating flexible 1D and 2D histograms in-situ that
+are written to disk in HDF5 format.
+Currently supported are
+
+- 1D and 2D histograms
+- binning by variable or coordinate (x1, x2, x3 and radial distance)
+- counting samples and or summing a variable  
+- weighting by volume and/or variable
+
+The output format follows ``numpy`` convention, so that plotting data
+with Python based machinery should be straightfoward (see example below).
+In general, histograms are calculated using inclusive left bin edges and
+data equal to the rightmost edge is also included in the last bin.
+
+A ``<parthenon/output*>`` block containing one simple and one complex
+example might look like::
+
+   <parthenon/output8>
+   file_type = histogram # required, sets the output type
+   dt = 1.0              # required, sets the output interval
+   num_histograms = 2    # required, specifies how many histograms are defined in this block
+
+   # 1D histogram ("standard", i.e., counting occurance in bin)
+   hist0_ndim = 1
+   hist0_x_variable = advected
+   hist0_x_variable_component = 0
+   hist0_x_edges_type = log
+   hist0_x_edges_num_bins = 10
+   hist0_x_edges_min = 1e-9
+   hist0_x_edges_max = 1e0
+   hist0_binned_variable = HIST_ONES
+
+   # 2D histogram of volume weighted variable according to two coordinates
+   hist1_ndim = 2
+   hist1_x_variable = HIST_COORD_X1
+   hist1_x_edges_type = list
+   hist1_x_edges_list = -0.5, -0.25, 0.0, 0.25, 0.5
+   hist1_y_variable = HIST_COORD_X2
+   hist1_y_edges_type = list
+   hist1_y_edges_list = -0.5, -0.1, 0.0, 0.1, 0.5
+   hist1_binned_variable = advected
+   hist1_binned_variable_component = 0
+   hist1_weight_by_volume = true
+   hist1_weight_variable = one_minus_advected_sq
+   hist1_weight_variable_component = 0
+
+with the following parameters
+
+- ``num_histograms=INT``
+   The number of histograms defined in this block.
+   All histogram definitions need to be prefix with ``hist#_`` where ``#`` is the
+   histogram number starting to count from ``0``.
+   All histograms will be written to the same output file with the "group" in the
+   output corresponding to the histogram number.
+- ``hist#_ndim=INT`` (either ``1`` or ``2``)
+   Dimensionality of the histogram.
+- ``hist#_x_variable=STRING`` (variable name or special coordinate string ``HIST_COORD_X1``, ``HIST_COORD_X2``, ``HIST_COORD_X3`` or ``HIST_COORD_R``)
```

Comment:
Hmm, not quite. I might need to update the documentation (I'm happy for suggestions to make it more clear).

When I wrote this/chose the names, I thought about it in terms of plots.
So for both 1D and 2D histograms the "x" variable is the one on the bottom axis, i.e., always used.
Then the "binned" variable is the "data" variable to be shown (or put into "bins"), i..e, for a 1D histogram this variable will correspond to the line (i.e., the y values) and for the 2D histogram this will be the "z" dim.
The "y" variable itself is only used for 2D histograms and is the variable on the left/y axis.

I realize this might be/is confusing, though I didn't come up with better naming scheme that works for 1D and 2D histograms while not changing the meaning of an input parameter.

Reply:
Hmm I'm still not sure I understand. Maybe a few examples would help me reason through it. In 1D, I often want to compute something like the top panel of this figure:
![image](https://github.com/parthenon-hpc-lab/parthenon/assets/1691313/c61e4af5-de49-48eb-8834-3695ee7a8636)
Note that each "curve" is basically "mass in a bin" vs bins of Ye. What variables should I pick in your framework to make this happen? My intuition would be `hist_x_variable` set to `Ye`, `hist_binned_variable` set to density, and volume-weighting turned on... is that right?

Reply:
Exactly right.
And to add a 2D example,
![image](https://github.com/parthenon-hpc-lab/parthenon/assets/22685568/ab5c931a-06f4-41fc-a353-b74533fffd85)
Here, the x variable is radius, the y variable plasma beta, and the "data" is again mass (calculated from binning the density field and apply volume weighting).

Reply:
Ok thanks for the explanation. This makes sense to me---although it might be useful to embed some of these examples in the docs to clarify. I think my head was stuck on what `matplotlib` requests of you when you build a histogram, which is slightly different, as it conflates some of these variables.

Reply:
I guess I don't know for sure---but they're from published works right? It seems like it should be ok in the context of documentation? (Mine is from DOI 10.3847/2041-8213/acba16)

Reply:
I now added two examples.